app/javascript/packs/contact-form-navigation-alert.js

```diff
@@ -0,0 +1,23 @@
+$(() => {
+   $("body").attr("data-turbolinks", false);
+
+   const formElement = $("form");
+   const originalState = formElement.serialize();
+   formElement.on("submit", onSubmit)
+
+   function shouldNavigateDirectly() {
+      const currentState = $("form").serialize();
+
+      $("*").blur();
+      return currentState === originalState ? undefined : false;
```

Comment:
I don't think this would ever evaluate to `true` because `===` between objects checks if they are the same piece of memory, rather than comparing their values.

Reply:
The result of the serialization is a string. When we use === for strings in JavaScript, we compare value and type. So, if the strings are equal by value, this will return true.

Quick snippet from the console:
![image](https://user-images.githubusercontent.com/63452375/80496360-c523f800-8971-11ea-8224-1593ddb7eec6.png)

Reply:
oh! I assumed it serialised to an object